internal/types/staticct/staticct.go

```diff
@@ -112,6 +112,61 @@ func (t *EntryBundle) UnmarshalText(raw []byte) error {
 	return nil
 }
 
+// ExtractTimestampFromBundle extracts the timestamp from the Nth entry in the provided serialised entry bundle.
+//
+// This implementation attempts to avoid any unecessary allocation or parsing other than whatever is
+// necessary to skip over uninteresting bytes to find the requested ExtractTimestampFromBundle.
+func ExtractTimestampFromBundle(ebRaw []byte, N uint64) (uint64, error) {
+	s := cryptobyte.String(ebRaw)
+
+	i := uint64(0)
+	for len(s) > 0 {
+		var timestamp uint64
+		var entryType uint16
+		if !s.ReadUint64(&timestamp) || !s.ReadUint16(&entryType) || timestamp > math.MaxInt64 {
+			return 0, fmt.Errorf("invalid data tile")
```

Comment:
Nit: include the index in the error for easier debugging if need be.

Reply:
Done, and I've also updated `CTStorage.dedupFuture` to wrap with bundle coords.